builtins/amp-ad.js

```diff
@@ -227,7 +232,7 @@ export function installAd(win) {
       // it is visible.
       if (inViewport) {
         this.unlistenViewportChanges_ =
-            this.getViewport().onChanged(this.sendAdIntersection_.bind(this));
+            this.getViewport().onScroll(this.sendAdIntersection_.bind(this));
```

Comment:
It would.

I definitely don't want to do only viewportChange.

If you think it would be too spammy I could throttle it to one event per 3 frames. I checked in Chrome and that is roughly how often I was getting scroll events anyway.

At some point I will also need to make a viewportCallback variant that uses onScroll instead of onChange.